docs/details/key-skew.md

```diff
@@ -1,43 +1,39 @@
 
-1. Processing partitions are unbalanced by a magnitude then the largest partition becomes the bottleneck
+1. Usually caused during a transformation when the data in one partition ends up being a lot larger than the others.
 
+2. If processing partitions are unbalanced by an order of magnitude, then the largest partition becomes the bottleneck.
+TODO: Is (2) intended to mean partitions created BEFORE a transformation? i.e. do we need both (1) and (2) here?
 
 How to identify skew
 
-1. If one task took much longer to complete than the other tasks, it's usually a sign of Skew. On the Spark UI under Summary Metrics for completed tasks if the Max duration is higher by a significant magnitude from the Median it usually represents Skew, e.g.:
+1. If one task took much longer to complete than the other tasks, it's usually a sign of skew. On the Spark UI --> Summary Metrics --> Completed Tasks, if the Max task duration is higher by a significant magnitude from the Median task duration, this likely represents skew, e.g.:
 ![Key-Skew-Spark-UI](../imgs/spark-skewed.png)
 
 Things to consider
 
-1. Mitigating skew has a cost (e.g. repartition) hence its ignorable unless the duration or input size is significantly higher in magnitude severely impacting job time
+1. Mitigating skew has a cost (e.g. repartition) hence skew is ignorable unless the duration or input size is significantly higher in magnitude severely impacting job time.
 
 
 Mitigation strategies
 
-1. Increasing executor memory to prevent OOM exceptions -> This a short-term solution if you want to unblock yourself but does not address the underlying issue. Sometimes this is not an option when you are already running at the max memory settings allowable. 
+1. Increasing executor memory to prevent OOM exceptions -> This a short-term strategy if you want to unblock yourself but does not address the underlying issue. Sometimes this approach is not an option when you are already running at the max memory settings allowable. Using this strategy, executors processing the majority of data partitions are over-allocated, potentially earning your job the title "resource-hog." So increase executor memory as a discovery and troubleshooting strategy rather than a "solution."
 
-1. Salting is a way to balance partitions by introducing a salt/dummy key for the skewed partitions. Here is a sample workbook and an example of salting in content performance show completion pipeline, where the whole salting operation is parametrized with a JOIN_BUCKETS variable which helps with maintenance of this job.
+2. Salting is a way to balance partitions by introducing a salt/dummy key for the skewed partitions. Here is a sample workbook and an example of salting in content performance show completion pipeline, where the whole salting operation is parametrized with a JOIN_BUCKETS variable which helps with maintenance of this job.
 ![Spark-Salted-UI](../imgs/spark-salted.png)
```

Comment:
What is the link to the workbook? Link to example? Sounds very useful!

Reply:
oooh good point, let me find that.